File: myself/src/main/java/com/xxd/myself/intdef/MyTest.java
package com.xxd.myself.intdef;

/**
 * author : xxd
 * date   : 2023/12/3
 * desc   :
 */
public class MyTest {


    public static void main(String[] args) {
//        setNumber(3);  不能直接使用数字了
        setNumber(DomainA.Type1);
    }

    public static void setNumber(@MyAnnotation int i){

    }
}

File: myself/src/main/java/com/xxd/myself/intdefks/MyTest.kt
package com.xxd.myself.intdefks

import com.xxd.myself.intdef.DomainA

/**
 *    author : xxd
 *    date   : 2023/12/3
 *    desc   :
 */
class MyTest {

    private fun fun1(){
//        setDomain(1)
        setDomain(DomainB.type3)
    }

    private fun setDomain(@MyAnnotation i: Int){

    }
}